"make renode" fails first time, works second time.  Possible race condition.

When I run the following:
```
make clean
make renode
```
...I get the following error:
```
  LD       software.elf
  OBJDUMP  software.elf.dis
  OBJDUMP  software.elf.sym
  OBJCOPY  software.bin
  IHEX     software.ihex
make[1]: Leaving directory '/home/tim/alanvgreen/CFU-Playground/proj/proj_template_v/build'
/home/tim/alanvgreen/CFU-Playground/soc/build/digilent_arty.proj_template_v/csr.csv not found
Running interactively under renode
/bin/cp -a /home/tim/alanvgreen/CFU-Playground/proj/proj_template_v/build/software.elf /home/tim/alanvgreen/CFU-Playground/proj/proj_template_v/renode/
pushd /home/tim/alanvgreen/CFU-Playground/proj/proj_template_v/build/renode/ && /home/tim/alanvgreen/CFU-Playground/third_party/renode/renode -e "s @digilent_arty.resc" && popd
/bin/bash: line 0: pushd: /home/tim/alanvgreen/CFU-Playground/proj/proj_template_v/build/renode/: No such file or directory
make: *** [../proj.mk:168: renode] Error 1
```

If I run `make renode` again (without doing a make clean), it then works (the two Renode windows come up).

I suspect that somewhere in our make commands, a parallel make is specified, and the dependencies are not well specified.  In particular, the first error is `/home/tim/alanvgreen/CFU-Playground/soc/build/digilent_arty.proj_template_v/csr.csv not found`, but this file **does** exist when I check.   So I think the consumer of this file tries to execute before (or in parallel with) the producer.

--- a/.github/includes/actions/run-in-renode/action.yml
+++ b/.github/includes/actions/run-in-renode/action.yml
@@ -40,7 +40,7 @@ runs:
       run:
         pwd && source environment && cd proj/${{ inputs.proj-name }} &&
         make ${{ inputs.build-params }} PLATFORM=${{ inputs.platform }} TARGET=${{ matrix.target }} -j8 software &&
-        make renode-scripts TARGET=${{ matrix.target }}
+        make renode-scripts ${{ inputs.build-params }} PLATFORM=${{ inputs.platform }} TARGET=${{ matrix.target }}
 
     - name: Run tests
       timeout-minutes: 5

--- a/.github/workflows/test-projects.yml
+++ b/.github/workflows/test-projects.yml
@@ -77,7 +77,7 @@ jobs:
     - name: Setup environment
       run: bash scripts/setup -ci
     - name: Build sample & generate Renode scripts
-      run: pwd && source environment && cd proj/proj_template && make  PLATFORM=common_soc TARGET=${{ matrix.target }} -j8 software && make renode-scripts TARGET=${{ matrix.target }}
+      run: pwd && source environment && cd proj/proj_template && make  PLATFORM=common_soc TARGET=${{ matrix.target }} -j8 software && make renode-scripts  PLATFORM=common_soc TARGET=${{ matrix.target }}
     - name: Run tests
       timeout-minutes: 5
       uses: antmicro/renode-actions/test-in-renode@main
@@ -122,7 +122,7 @@ jobs:
     - name: Setup environment
       run: bash scripts/setup -ci
     - name: Build sample & generate Renode scripts
-      run: pwd && source environment && cd proj/mnv2_first && make SW_ONLY=1 PLATFORM=common_soc TARGET=${{ matrix.target }} -j8 software && make renode-scripts TARGET=${{ matrix.target }}
+      run: pwd && source environment && cd proj/mnv2_first && make SW_ONLY=1 PLATFORM=common_soc TARGET=${{ matrix.target }} -j8 software && make renode-scripts SW_ONLY=1 PLATFORM=common_soc TARGET=${{ matrix.target }}
     - name: Run tests
       timeout-minutes: 5
       uses: antmicro/renode-actions/test-in-renode@main
@@ -166,7 +166,7 @@ jobs:
     - name: Setup environment
       run: bash scripts/setup -ci
     - name: Build sample & generate Renode scripts
-      run: pwd && source environment && cd proj/mnv2_first && make SW_ONLY=1 PLATFORM=hps TARGET=${{ matrix.target }} -j8 software && make renode-scripts TARGET=${{ matrix.target }}
+      run: pwd && source environment && cd proj/mnv2_first && make SW_ONLY=1 PLATFORM=hps TARGET=${{ matrix.target }} -j8 software && make renode-scripts SW_ONLY=1 PLATFORM=hps TARGET=${{ matrix.target }}
     - name: Run tests
       timeout-minutes: 5
       uses: antmicro/renode-actions/test-in-renode@main

--- a/proj/proj.mk
+++ b/proj/proj.mk
@@ -162,20 +162,16 @@ endif
 TARGET_REPL := $(BUILD_DIR)/renode/$(TARGET)_generated.repl
 
 .PHONY:	renode 
-renode: $(SOFTWARE_ELF) renode-scripts
+renode: renode-scripts
 	@echo Running interactively under renode
 	$(COPY) $(SOFTWARE_ELF) $(PROJ_DIR)/renode/
 	pushd $(PROJ_DIR)/build/renode/ && $(RENODE_DIR)/renode -e "s @$(TARGET).resc" && popd
 
 .PHONY: renode-scripts
-renode-scripts:
-ifneq ("$(wildcard $(SOC_BUILD_DIR)/csr.csv)","")
+renode-scripts: $(SOFTWARE_ELF)
 	@mkdir -p $(BUILD_DIR)/renode
 	$(CFU_ROOT)/scripts/generate_renode_scripts.py $(SOC_BUILD_DIR)/csr.csv $(TARGET) $(BUILD_DIR)/renode/ --repl $(TARGET_REPL)
 	@echo Generating Renode scripts finished
-else
-	@echo $(SOC_BUILD_DIR)/csr.csv not found
-endif
 
 .PHONY: clean
 clean:

